h2o-core/src/main/java/water/H2O.java

```diff
@@ -1324,6 +1324,9 @@ public static String DEFAULT_FLOW_DIR() {
    * It is updated also when a new client appears. */
   private static HashSet<H2ONode> STATIC_H2OS = null;
```

Comment:
should this be a non-blocking hash set? We can be modifying this from multiple threads, is HashSet.add thread safe?